Packs/PrismaSaasSecurity/Integrations/SaasSecurityEventCollector/SaasSecurityEventCollector.py

```diff
@@ -0,0 +1,254 @@
+        Get only a single event log - used only for the test-module to verify log-access api scope.
+        """
+        return self.http_request(
+            'GET',
+            url_suffix='/api/v1/log_events',
+            resp_type='response',
+            ok_codes=[200, 204],
+        )
+
+
+def is_token_expired(token_initiate_time: float, token_expiration_seconds: float) -> bool:
+    """
+    Check whether a token has expired. a token considered expired if it has been reached to its expiration date in
+    seconds minus a minute.
+
+    for example ---> time.time() = 300, token_initiate_time = 240, token_expiration_seconds = 120
+
+    300.0001 - 240 < 120 - 60
+
+    Args:
+        token_initiate_time (float): the time in which the token was initiated in seconds.
+        token_expiration_seconds (float): the time in which the token should be expired in seconds.
+
+    Returns:
+        bool: True if token has expired, False if not.
+    """
+    return time.time() - token_initiate_time >= token_expiration_seconds - 60
+
+
+def validate_limit(limit: int):
+    """
+    Validate that the limit/max fetch is a number divisible by the MAX_EVENTS_PER_REQUEST (100).
+    """
+    if limit % MAX_EVENTS_PER_REQUEST != 0:
+        raise DemistoException(f'fetch limit parameter should be divisible by {MAX_EVENTS_PER_REQUEST}')
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client):
+    """
+    Testing we have a valid connection to Saas-Security.
+
+    The reason not testing fetching events is that if we fetch it from the api we will lose those requests.
+    """
+    response = client.get_event_request()
+    if response.status_code in (200, 204):
+        return 'ok'
+
+
+def saas_security_get_events_command(client: Client, args: Dict) -> Union[str, CommandResults]:
+    """
+    Fetches events from the saas-security queue and return them to the war-room.
+    in case should_push_events is set to True, they will be also sent to XSIAM.
+    """
+    limit = arg_to_number(args.get('limit')) or 100
+    validate_limit(limit=limit)
+    should_push_events = argToBoolean(args.get('should_push_events'))
+
+    if events := fetch_events(client=client, max_fetch=limit):
+        if should_push_events:
+            send_events_to_xsiam(events=events, vendor='PaloAltoNetworks', product='SaasSecurity')
+        return CommandResults(
+            readable_output=tableToMarkdown(
+                'SaaS Security Logs',
+                events,
+                headers=['log_type', 'item_type', 'item_name', 'timestamp', 'serial'],
+                headerTransform=underscoreToCamelCase,
+                removeNull=True
+            ),
+            raw_response=events,
+            outputs=events,
+            outputs_key_field=['timestamp', 'log_type, item_name, item_type'],
+            outputs_prefix='SaasSecurity.Event'
+        )
+    return 'No events were found.'
+
+
+def fetch_events(client: Client, max_fetch: int) -> List[Dict]:
```

Comment:
lets rename to fetch_events_from_saas_security, to be clear this isnt the command fn